src/constants.rs

```diff
@@ -35,8 +39,83 @@ pub const TYPE_ID_CODE_HASH: H256 = h256!("0x545950455f4944");
 
 pub const SIGHASH_TYPE_HASH: H256 =
     h256!("0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8");
-pub const MULTISIG_TYPE_HASH: H256 =
-    h256!("0x5c5069eb0857efc65e1bca0c07df34c31663b3622fd3876c876320fc9634e2a8");
+
+pub const GENESIS_BLOCK_HASH_MAINNET: H256 =
+    h256!("0x92b197aa1fba0f63633922c61c92375c9c074a93e85963554f5499fe1450d0e5");
+
+pub const GENESIS_BLOCK_HASH_TESTNET: H256 =
+    h256!("0x10639e0895502b5688a6be8cf69460d76541bfa4821629d86d62ba0aae3f9606");
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum MultisigScript {
+    /// Multisig Script deployed on Genesis Block
+    /// https://explorer.nervos.org/script/0x5c5069eb0857efc65e1bca0c07df34c31663b3622fd3876c876320fc9634e2a8/type
+    Legacy,
+
+    /// Latest multisig script
+    /// https://explorer.nervos.org/script/0x36c971b8d41fbd94aabca77dc75e826729ac98447b46f91e00796155dddb0d29/data1
+    V1,
+}
+
+impl MultisigScript {
+    pub const fn script_id(&self) -> ScriptId {
+        match self {
+            MultisigScript::Legacy => ScriptId::new_type(h256!(
+                "0x5c5069eb0857efc65e1bca0c07df34c31663b3622fd3876c876320fc9634e2a8"
+            )),
+            MultisigScript::V1 => ScriptId::new_data1(h256!(
+                "0x36c971b8d41fbd94aabca77dc75e826729ac98447b46f91e00796155dddb0d29"
+            )),
+        }
+    }
+
+    pub const fn dep_group(&self, network: NetworkType) -> (H256, u32) {
+        match self {
+            MultisigScript::Legacy => match network {
+                NetworkType::Mainnet => (
+                    h256!("0x71a7ba8fc96349fea0ed3a5c47992e3b4084b031a42264a018e0072e8172e46c"),
+                    1,
+                ),
+                NetworkType::Testnet => (
+                    h256!("0xf8de3bb47d055cdf460d93a2a6e1b05f7432f9777c8c474abf4eec1d4aee5d37"),
+                    1,
+                ),
+                NetworkType::Staging => todo!(),
```

Comment:
We can use env variables as a temporarily solution to get the configurations for other environments.